src/index.ts

```diff
@@ -1,92 +1,166 @@
 class School {
     // implement 'add area', 'remove area', 'add lecturer', and 'remove lecturer' methods
+    _areas: string[] = [];
+    _lecturers: string[] = []; // Name, surname, position, company, experience, courses, contacts
 
-    _areas = [];
-    _lecturers = []; // Name, surname, position, company, experience, courses, contacts
+    addArea(area: string): void {
+        this._areas.push(area);
+    }
+
+    removeArea(area: string): void {
+        this._areas = this._areas.filter((element: string): boolean => element !== area);
+    }
+
+    addLecturer(lecturer: string): void {
+        this._lecturers.push(lecturer);
+    }
 
-    get areas() {
-        this._areas;
+    removeLecturer(lecturer: string): void {
+        this._lecturers = this._lecturers.filter((element: string): boolean => element !== lecturer);
     }
 
-    get lecturers() {
-        this._lecturers;
+
+    get areas(): string[] {
+        return this._areas;
+    }
+
+    get lecturers(): string[] {
+        return this._lecturers;
     }
 }
 
 class Area {
     // implement getters for fields and 'add/remove level' methods
-    _levels = [];
-    _name;
+    _levels: number[] = [];
+    _name: string;
 
-    constructor(name) {
+    constructor(name: string) {
         this._name = name;
     }
+
+    addLevel(level: number): void {
+        this._levels.push(level);
+    }
+
+    removeLevel(level: number): void {
+        this._levels = this._levels.filter((element: number): boolean => element !== level);
+    }
+
+    get levels(): number[] {
+        return this._levels;
+    }
+
+    get name(): string {
+        return this._name;
+    }
 }
 
 class Level {
     // implement getters for fields and 'add/remove group' methods
+    _groups: string[] = [];
+    _name: string;
+    _description: string;
 
-    _groups;
-    _name;
-
-    constructor(name, description) {
-        this.name = name;
+    constructor(name: string, description: string) {
+        this._name = name;
         this._description = description;
     }
+
+    addGroup(group: string): void {
+        this._groups.push(group);
+    }
+
+    removeGroup(group: string) {
```

Comment:
Do not forget to add type annotation for the return function value explicitly.

Reply:
Done